### DISCUSSION TITLE
How to implement an LSTM on nodes in a batch of graphs?

### DISCUSSION THREAD

**Discussion by Bowen-n** [pyg-team/pytorch_geometric#4888] [Q&A] [answered]
Given a Batch, besides a common GNN, I also want to implement an **LSTM** on the nodes of graphs.

e.g., for each graph, the order of the node is as their orders in Batch (Data), and these nodes are fed into an LSTM to generate new node embeddings.

Besides, graphs in a batch may contain **different** numbers of nodes.

**Comment by rusty1s** [comment, marked as answer]
You can utilize `utils.to_dense_batch`, which will generate a feature tensor of shape `[batch_size, num_max_nodes, num_features]`, which you can then input into an `LSTM`.
